### ISSUE TITLE
Building on OSX fails...

### ISSUE THREAD

**Issue by Robert-M-Muench** [CyberShadow/Digger#86]
```
digger: Running: 'make' '-f' 'posix.mak' 'MODEL=64' 'HOST_DC=/Users/Shared/develop/d-language/Digger-OSX/work/dl/dmd-2.079.0/dmd2/osx/bin/dmd'
posix.mak:75: ========== Use HOST_DMD instead of HOST_DC ==========
/Users/Shared/develop/d-language/Digger-OSX/work/dl/dmd-2.079.0/dmd2/osx/bin/dmd -of../generated/build -g build.d
dyld: lazy symbol binding failed: Symbol not found: _dyld_enumerate_tlv_storage
  Referenced from: /Users/Shared/develop/d-language/Digger-OSX/work/dl/dmd-2.079.0/dmd2/osx/bin/dmd
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: _dyld_enumerate_tlv_storage
  Referenced from: /Users/Shared/develop/d-language/Digger-OSX/work/dl/dmd-2.079.0/dmd2/osx/bin/dmd
  Expected in: /usr/lib/libSystem.B.dylib
```

I don't know what /usr/lib/libSystem.B.dylib is, where it comes from nor why there is a symbol missing.

**Comment by CyberShadow** [comment]
Apple keeps breaking old binaries in newer versions of their operating system. Try playing with the bootstrap options, perhaps it's possible to build starting with the last C++ version.

I don't own any Apple hardware, and due to the increasing difficulty of installing the newer versions of Apple's operating systems in virtual machines, I may not be able to keep maintaining support for them in Digger.

**Comment by CyberShadow** [comment]
> Try playing with the bootstrap options, perhaps it's possible to build starting with the last C++ version.

I set up an OSX VM and have been looking at this problem the last few days.

The underlying issue is https://issues.dlang.org/show_bug.cgi?id=20019. Old D versions depended on `_dyld_enumerate_tlv_storage`, which was removed in macOS 10.15.

Research notes follow:

### Build matrix

Results of trying to naively build every recent D version with every recent D version with Digger:

    host ->   / v2.06x \/ v2.07x \/ v2.08x \/ v2.09x \
    target v: 0123456789012345678901234567890123456789
    v2.060  : ...................................
    v2.061  : ...................................
    v2.062  : ...................................
    v2.063  : ...................................
    v2.064  : ...................................
    v2.065.0: ...................................
    v2.066.0: ...................................
    v2.067.0: ...................................
    v2.068.0: ...................................
    v2.069.0: ...................................
    v2.070.0: ...................................
    v2.071.0: ...................................
    v2.072.0: ...................................
    v2.073.0: ...................................
    v2.074.0: ...................................
    v2.075.0: ...................................
    v2.076.0: ...................................
    v2.077.0: ...................................
    v2.078.0: ...................................
    v2.079.0: ...................................
    v2.080.0: ...................................
    v2.081.0: ...................................
    v2.082.0: ...................................
    v2.083.0: ...................................
    v2.084.0: ...................................
    v2.085.0: ...................................
    v2.086.0: ...................................
    v2.087.0: ...................................
    v2.088.0: ............................yyyy...
    v2.089.0: ............................yyyy...
    v2.090.0: ............................yyyy...
    v2.091.0: ............................yyyy...
    v2.092.0: ............................yyyyyyy
    v2.093.0: ............................yyyyyyy
    v2.094.0: ............................yyyyyyy


<a id="orgb206ff4"></a>

### `_dyld_enumerate_tlv_storage` error with precompiled D versions

    dyld: lazy symbol binding failed: Symbol not found: _dyld_enumerate_tlv_storage
      Referenced from: /Users/vladimir/work/Digger/work/dl/dmd-2.073.0/dmd2/osx/bin/dmd
      Expected in: /usr/lib/libSystem.B.dylib
    
    dyld: Symbol not found: _dyld_enumerate_tlv_storage
      Referenced from: /Users/vladimir/work/Digger/work/dl/dmd-2.073.0/dmd2/osx/bin/dmd
      Expected in: /usr/lib/libSystem.B.dylib

-   first failing version is 2.073
-   first fixed   version is 2.088

#### `Undefined symbols for architecture x86_64`

Compilers before 2.073.0 can't build anything anyway, the error now comes when trying to link the resulting executable:
    
    Undefined symbols for architecture x86_64:
      "_dyld_enumerate_tlv_storage", referenced from:
          __d_dyld_getTLSRange in libphobos2.a(osx_tls.o)


<a id="org2dddb5a"></a>

### `-lstdc++` error

    CC=c++ /Users/vladimir/work/Digger/work/dl/dmd-2.088.0/dmd2/osx/bin/dmd idgen.d
    clang: warning: libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9 [-Wdeprecated]
    ld: library not found for -lstdc++
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    Error: linker exited with status 1

-   depends on version being built
-   fixed in 2.079


<a id="org2d52010"></a>

### Shared static deprecation

v2.092+ can't build v2.091 and older because ``"ERROR: dmd/tokens.d(710): Deprecation: `shared static` constructor can only be of D linkage"``.

**Comment by CyberShadow** [comment]
Should be fixed in Digger v3.0.6.
